src/credential-status/status-token.ts

```diff
@@ -0,0 +1,159 @@
+import AbortController from 'abort-controller'
+import fetch from 'node-fetch'
+import { cborDecode, cborEncode } from '../cbor'
+import { Tag } from '../cbor/cbor-x'
+import type { CoseKey } from '../cose'
+import { CoseType, CoseTypeToTag, Mac0, Sign1 } from '../cose'
+import { CWT } from '../cwt'
+import type { StatusArray } from './status-array'
+import { StatusList } from './status-list'
+
+interface CWTStatusTokenOptions {
+  claimsSet: {
+    statusArray: StatusArray
+    aggregationUri?: string
+    expirationTime?: number
+    timeToLive?: number
+  }
+  type: CoseType
+  key: CoseKey
+}
+
+interface CWTStatusTokenVerifyOptions {
+  token: Uint8Array
+  key?: CoseKey
+}
+
+interface CWTStatusVerifyOptions extends CWTStatusTokenVerifyOptions {
+  index: number
+  expectedStatus: number
+}
+
+enum CwtProtectedHeaders {
+  TYPE = 16,
+}
+
+enum CwtStatusListClaims {
+  StatusListUri = 2,
+  ExpirationTime = 4,
+  IssuedAt = 6,
+  StatusList = 65533,
+  TimeToLive = 65534,
+}
+
+export class CWTStatusToken {
+  static async build(options: CWTStatusTokenOptions): Promise<Uint8Array> {
+    const cwt = new CWT()
+    cwt.setHeaders({
+      protected: {
+        [CwtProtectedHeaders.TYPE]: 'application/statuslist+cwt',
+      },
+    })
+
+    const claims: { [key: number]: string | number | Uint8Array } = {
+      [CwtStatusListClaims.StatusListUri]: 'https://example.com/statuslist', // Where the status list is going to be hosted
+      [CwtStatusListClaims.IssuedAt]: Math.floor(Date.now() / 1000),
+      [CwtStatusListClaims.StatusList]: StatusList.buildCborStatusList({
+        statusArray: options.claimsSet.statusArray,
+        aggregationUri: options.claimsSet.aggregationUri,
+      }),
+    }
+    if (options.claimsSet.expirationTime) {
+      claims[CwtStatusListClaims.ExpirationTime] = options.claimsSet.expirationTime
+    }
+    if (options.claimsSet.timeToLive) {
+      claims[CwtStatusListClaims.TimeToLive] = options.claimsSet.timeToLive
+    }
+
+    cwt.setClaims(claims)
+    return cborEncode(new Tag(await cwt.create({ type: options.type, key: options.key }), CoseTypeToTag[options.type]))
+  }
+
+  static async verifyStatusToken(options: CWTStatusTokenVerifyOptions): Promise<boolean> {
+    const cwt = cborDecode(options.token) as Sign1 | Mac0
+
+    const type = cwt.protectedHeaders.headers?.get(String(CwtProtectedHeaders.TYPE))
+    if (!type || type !== 'application/statuslist+cwt') {
```

Comment:
I think we should also extract the `application/statuslist+cwt` into an Enum/const so we can reuse it